### PR TITLE
AIs can now change their core lights!

### PR DESCRIPTION
**Pull request by Kmc2000**
### Intent of your Pull Request

Ais can now make their floor light up different colours, comes in a few colours and also PARTY mode

To see it in action, go here:
https://drive.google.com/file/d/0B9VUbibptgF2ek9PNDV3Y3RERGM/view?usp=sharing

#### Changelog

:cl: Kmc2000
rscadd: AIs can now change their floor lights! check your AI commands tab.
tweak: Mappers, the AI's floor light command will only change /turf/open/floor/bluegrid/AI, not the regular one, keep that in mind.
/:cl:

### CODE REVIEW

**Comment by ShadowDeath6** [comment]
cool

I'd disable the move during the malf countdown

also party mode seems like it would lag a bit

**Comment by Kmc2000** [comment]
>countdown

Meh, could do but i dunno if it's a massive issue.

EDIT:
The main lag from party mode in the clip is my game having bad FPS from hosting locally

**Comment by krgii** [comment]
Looks good.

**Comment by Jc1469** [comment]
tbh AI has enough ways to kill people, irl with epilepsy seems a bit OP.

balance police do not approve.

**Comment by Kmc2000** [comment]
Balance police have no jurisdiction here, this case falls under the Federal Balance Institute's authority

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 1]
```diff
@@ -746,6 +746,11 @@ var/list/ai_list = list()
 	if(stat == 2)
 		return //won't work if dead
 
+	for(var/turf/open/floor/bluegrid/AI/A)
+		if(A.nuking == 1)
+			src << "ERROR lighting circuits overriden by outside forces, perhaps the self destruct sequence has been activated?."
```
Comment (ShadowDeath6) [inline code/modules/mob/living/silicon/ai/ai.dm]: ?.

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 2]
```diff
@@ -730,6 +731,48 @@ var/list/ai_list = list()
...
+		"green",
+		"yellow",
+		"pink",
+		"PARTY"
+		)
+
+	if(stat == 2)
+		return //won't work if dead
+
+	for(var/turf/open/floor/bluegrid/AI/A)
```
Comment (Alek2ander) [inline code/modules/mob/living/silicon/ai/ai.dm:749]: Iterating over everything in world = no
There's a list of nuke_tiles

File: code/game/gamemodes/nuclear/nuclearbomb.dm [hunk 3]
```diff
@@ -426,6 +426,8 @@ This is here to make the tiles around the station mininuke change when it's arme
 		for(var/N in nuke_tiles)
 			var/turf/open/floor/T = N
 			T.icon_state = (timing ? "rcircuitanim" : T.icon_regular_floor)
+		for(var/turf/open/floor/bluegrid/AI/A)
```
Comment (Alek2ander) [inline code/game/gamemodes/nuclear/nuclearbomb.dm:429]: Iterating over everything in world = no
There's a list of nuke_tiles

File: code/game/turfs/simulated/floor/misc_floor.dm [hunk 4]
```diff
@@ -59,6 +59,11 @@
 	icon = 'icons/turf/floors.dmi'
 	icon_state = "bcircuit"
 	floor_tile = /obj/item/stack/tile/plasteel
+	var/nuking = 0 //is a nuke active?
```
Comment (Alek2ander) [inline code/game/turfs/simulated/floor/misc_floor.dm:62]: Why are you storing redundant copies of the nuking state in a dozen floor tiles? Put that in ticker.

Actually, I'm pretty sure this is already somewhere in ticker, just use that.

**Review by Alek2ander — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
atleast your code isn't full of whitespace this time

**Comment by drderp3635** [comment]
would this change the color of the tiles in the xeno kill chamber?

**Comment by matskuman5** [comment]
I like it, but

Can't you just make it check for tiles in the AI core and upload area? To prevent what Alek and Drderp brought up?

**Comment by matskuman5** [comment]
>Add more commits by pushing to the master branch on Kmc2000/yogstation-1.

fucking UNCLEAN MASTER BRANCH REEEEEEEEEEEEE

**Comment by Kmc2000** [comment]
These are seperate tiles that ONLY affect the AI's core

**Comment by Kmc2000** [comment]
moving this from master branch

**Pull request closed without merging**